packages/kbn-test/src/auth/saml_auth.ts

```diff
@@ -102,6 +112,42 @@ export const createCloudSession = async (params: CreateSamlSessionParams) => {
     };
   };
 
+  let attemptsLeft = retryParams.attemptsCount;
+  while (attemptsLeft > 0) {
+    try {
+      sessionResponse = await axios.request(requestConfig(cloudLoginUrl));
+      if (sessionResponse?.status !== 200) {
+        throw new Error(
+          `Failed to create the new cloud session: 'POST ${cloudLoginUrl}' returned ${sessionResponse?.status}`
+        );
+      }
+      const token = sessionResponse?.data?.token as string;
+      if (!token) {
+        throw new Error(
+          `Failed to create the new cloud session: token is missing in response data\n${JSON.stringify(
+            sessionResponse?.data
+          )}`
+        );
+      }
+      return token;
+    } catch (ex) {
+      cleanException(cloudLoginUrl, ex);
+
+      attemptsLeft--;
+      if (attemptsLeft > 0) {
+        // log only error message
+        log.error(`${ex.message}\nWaiting ${retryParams.attemptDelay} ms before the next attempt`);
+        await new Promise((resolve) => setTimeout(resolve, retryParams.attemptDelay));
+      } else {
+        log.error(
+          `Failed to create the new cloud session with ${retryParams.attemptsCount} attempts`
+        );
+        // throw original error with stacktrace
+        throw ex;
+      }
+    }
+  }
+
   try {
     sessionResponse = await axios.request(requestConfig(cloudLoginUrl));
   } catch (ex) {
```

Comment:
I think we don't need this bit anymore?

Reply:
omg, accidentally applied stashed "old" logic. Thank you!